### ISSUE TITLE
iceprod2 - task resources (Trac #843)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#837]
Tasks need to specify what resources they require.

Need some sort of matching system between node resources and task requirements.

Node resources are stubbed out here:
[http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/sandbox/dschultz/iceprod2/iceprod/core/util.py#L28 iceprod/core/util.py#L28]

Migrated from https://code.icecube.wisc.edu/ticket/843
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "Tasks need to specify what resources they require.\n\nNeed some sort of matching system between node resources and task requirements.\n\nNode resources are stubbed out here:\n[http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/sandbox/dschultz/iceprod2/iceprod/core/util.py#L28 iceprod/core/util.py#L28]", 
    "reporter": "david.schultz", 
    "cc": "ddelventhal", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "iceprod2 - task resources", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2015-01-06T18:02:51", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150106T18:03:38: david.schultz commented:

> I have local modifications to node resources that should make this easier.  Will hopefully commit in the next few days.

**Comment by icecube-trac** [comment]
Trac update at 20150407T22:05:28: david.schultz commented:

> In changeset 131168:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="131168"
> mega commit:
> - the concept of a "resource" that tasks can use
>   TODO: not used by queueing yet.
> - global queueing framework
>   TODO: make sure local queueing and global queueing work together.
> - separate out dbmethods tests
> - fix up some failing tests
> 
> This partially addresses #837 and #838.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150504T15:56:02: david.schultz changed milestone from "" to "IceProd 2.0"

**Comment by icecube-trac** [comment]
Trac update at 20150831T19:06:02: david.schultz changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20151118T20:46:06: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz changed status from "accepted" to "closed"
* david.schultz commented:

> In changeset 139663:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="139663"
> resource matching for queueing. closes #837
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas commented:

> Milestone IceProd 2.0 deleted


* olivas changed milestone from "IceProd 2.0" to ""
